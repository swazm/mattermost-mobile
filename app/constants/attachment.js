// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_ATTACHMENT_FOOTER_LENGTH = 300;
