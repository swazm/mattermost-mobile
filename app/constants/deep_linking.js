// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    CHANNEL: 'channel',
    PERMALINK: 'permalink',
    OTHER: 'other',
};
