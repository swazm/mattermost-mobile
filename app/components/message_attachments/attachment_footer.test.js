// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {MAX_ATTACHMENT_FOOTER_LENGTH} from 'app/constants/attachment';
import AttachmentFooter from './attachment_footer';

import Preferences from 'mattermost-redux/constants/preferences';

describe('AttachmentFooter', () => {
    const baseProps = {
        text: 'This is the footer!',
        icon: 'https://images.com/image.png',
        theme: Preferences.THEMES.default,
    };

    test('it matches snapshot when no footer is provided', () => {
        const props = {
            ...baseProps,
            text: undefined,
            icon: undefined,
        };

        const wrapper = shallow(<AttachmentFooter {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('it matches snapshot when footer text is provided', () => {
        const props = {
            ...baseProps,
            icon: undefined,
        };

        const wrapper = shallow(<AttachmentFooter {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('it matches snapshot when only the footer icon is provided', () => {
        const props = {
            ...baseProps,
            text: undefined,
        };

        const wrapper = shallow(<AttachmentFooter {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('it matches snapshot when both footer and footer_icon are provided', () => {
        const wrapper = shallow(<AttachmentFooter {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('it matches snapshot when the footer is longer than the maximum length', () => {
        const props = {
            ...baseProps,
            text: 'a'.repeat(MAX_ATTACHMENT_FOOTER_LENGTH + 1),
        };

        const wrapper = shallow(<AttachmentFooter {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
