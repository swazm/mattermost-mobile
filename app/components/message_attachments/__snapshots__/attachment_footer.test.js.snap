// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentFooter it matches snapshot when both footer and footer_icon are provided 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 5,
    }
  }
>
  <Image
    key="footer_icon"
    source={
      Object {
        "uri": "https://images.com/image.png",
      }
    }
    style={
      Object {
        "height": 12,
        "marginRight": 5,
        "marginTop": 1,
        "width": 12,
      }
    }
  />
  <Text
    key="footer_text"
    style={
      Object {
        "color": "rgba(61,60,64,0.5)",
        "fontSize": 11,
      }
    }
  >
    This is the footer!
  </Text>
</View>
`;

exports[`AttachmentFooter it matches snapshot when footer text is provided 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 5,
    }
  }
>
  <Text
    key="footer_text"
    style={
      Object {
        "color": "rgba(61,60,64,0.5)",
        "fontSize": 11,
      }
    }
  >
    This is the footer!
  </Text>
</View>
`;

exports[`AttachmentFooter it matches snapshot when no footer is provided 1`] = `""`;

exports[`AttachmentFooter it matches snapshot when only the footer icon is provided 1`] = `""`;

exports[`AttachmentFooter it matches snapshot when the footer is longer than the maximum length 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 5,
    }
  }
>
  <Image
    key="footer_icon"
    source={
      Object {
        "uri": "https://images.com/image.png",
      }
    }
    style={
      Object {
        "height": 12,
        "marginRight": 5,
        "marginTop": 1,
        "width": 12,
      }
    }
  />
  <Text
    key="footer_text"
    style={
      Object {
        "color": "rgba(61,60,64,0.5)",
        "fontSize": 11,
      }
    }
  >
    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaâ€¦
  </Text>
</View>
`;
